### ISSUE TITLE
Cannot place text on one edge of pages rotated 90 or 270 degrees ("rect must be finite and not empty")

### ISSUE THREAD

**Issue by AlisterH** [pymupdf/PyMuPDF#1342]
Thanks for your guidance the other day on my other ticket.  I believe I have found a legitimate issue now, although I guess the culprit could be mupdf rather than pymupdf.

I have adapted the demo from https://github.com/pymupdf/PyMuPDF-Utilities/blob/master/demo/fitz-logo.py, attempting to place a logo in the same corner of the pdf regardless of page rotation.
I believe I am calculating the rects correctly, because if I draw rectangles to try to debug it looks right.

## To Reproduce
Running the attached script produces an error on the final page, which is rotated 270 degrees:
````
#  python3 logo_overlay.py input.pdf testlogo.pdf output.pdf
Rect(0.0, 0.0, 595.3200073242188, 841.9199829101562)
0
Rect(506.49479373753854, 0.0, 595.3200073242188, 35.5)

Rect(0.0, 0.0, 841.9199829101562, 595.3200073242188)
90
Rect(0.0, 0.0, 35.5, 88.82521358668023)

Rect(0.0, 0.0, 595.3200073242188, 841.9199829101562)
180
Rect(0.0, 806.4199829101562, 88.82521358668023, 841.9199829101562)

Rect(0.0, 0.0, 841.9199829101562, 595.3200073242188)
270
Rect(559.8200073242188, 753.094769323476, 595.3200073242188, 841.9199829101562)
Traceback (most recent call last):
  File "logo_overlay.py", line 84, in <module>
    overlay = True)
  File "/usr/local/lib/python3.7/dist-packages/fitz/utils.py", line 180, in show_pdf_page
    raise ValueError("rect must be finite and not empty")
ValueError: rect must be finite and not empty
````

## Expected behavior
Script should run without error, placing a red rectangle and "logo" text at the top right corner of each page.

## Debugging:
Comment out this block and there is no error - red rectangles are drawn as expected:
````
    xref = page.show_pdf_page(rect, src, 0,
                            reuse_xref = xref,
                            rotate = page.rotation, # rotate logo to match actual page orientation
                            overlay = True)
````
Comment out this line and inspect the results.  On the last page the logo text is tiny - it seems that when placing the text pymupdf thinks that rect is almost off the edge of the page:
````
        rect = fitz.Rect(pagerect[2] - srcrect[3], pagerect[3] - srcrect[2], pagerect[2], pagerect[3]) # comment out this line for debugging purposes
````
I wondered if there was something else I'd missed in the docs, and which perhaps needed to be added to the section on "Misplaced Item Insertions on PDF Pages".  But I think in recent versions I should be able to insert the "logo" using the same rect that works for drawing a rectangle.
## Your configuration
 - Ubuntu 19.10 running in WSL 1 on Windows 10
 - Python 3.7.5
 - PyMuPDF 1.19.1, installed from wheel.
 ## Example data
[working.zip](https://github.com/pymupdf/PyMuPDF/files/7421387/working.zip)

**Comment by JorjMcKie** [comment]
Resolved by version 1.19.2.

**Comment by shogo-nakano-desu** [comment]
This fix helps me a lot.
Thank you so much for your contribution.
